### DISCUSSION TITLE
Code-completion issues in IntelliJ

### DISCUSSION THREAD

**Discussion by tmtron** [josdejong/mathjs#2809] [Q&A] [answered]
As described on [Stackoverflow: Why does code-completion work on Stackblitz, but not in Intellij?](https://stackoverflow.com/questions/74026987/why-does-code-completion-work-on-stackblitz-but-not-in-intellij), code completion does not work for mathJs in IntelliJ.

* I think the reason for this is, that you have an [index.ts](https://github.com/josdejong/mathjs/blob/v11.3.0/types/index.ts) file and an [index.d.ts](https://github.com/josdejong/mathjs/blob/v11.3.0/types/index.d.ts) file which are not related to each other, despite the same name. 
  * So I think, you should **rename** the `index.ts` file to something else: maybe `main.ts`, etc.  
    When I rename `index.ts` to `main.ts` in `node_modules/types/` then code-completion in IntelliJ works
* I wonder why the [index.ts](https://github.com/josdejong/mathjs/blob/v11.3.0/types/index.ts) is even included in the npm bundle. What is it needed for? It seems include only some code, that is meant for the developers of the library, but not for the users of the lib.
  * If my assumption is correct, then the `index.ts` file should not be included in the npm bundle at all  
    When I delete the `index.ts` file, then code completion in IntelliJ works.

What do you think?

**Comment by gwhitney** [comment]
Thanks for fielding the stack exchange question, tmtron.

I agree that the current index.ts has no value for someone who is simply a client of the library. So, not bundling index.ts sounds fine to me. But I am not really familiar with the bundler or how to leave the index.ts out; maybe this is something someone else involved with the project (possibly @josdejong if he is not too swamped) can do easily?

**Comment by josdejong** [comment, marked as answer]
Thanks for bringing this up, good point. The `index.ts` file is only used as a unit test to verify that all type definitions work. It should indeed not be a part of the npm library. I think we the most neat solution is to move `index.ts` into the folder with unit tests. Alternatively, we could simply ignore the file in the npm package.

**Reply by josdejong** [reply]
I've created a PR fixing this in #2812. Unless there is feedback I'll merge it in a few days.

**Reply by josdejong** [reply]
I've published a fix now in `v11.3.1`

**Reply by tmtron** [reply]
> I've published a fix now in `v11.3.1`

Thanks, I can verify that code-completion works now.
